### PR TITLE
Support docker environment vars from v2

### PR DESCRIPTION
**Pull request by lox**
This adds support for everything in https://buildkite.com/docs/builds/docker-containerized-builds, but also displays warnings.

### CODE REVIEW

File: bootstrap/bootstrap.go [hunk 1]
```diff
@@ -96,26 +96,6 @@ func (b *Bootstrap) Start() int {
 	return exitStatus
 }
 
-// executeScript executes a script in a Shell, but the target is an interpreted script
```
Comment (lox) [inline bootstrap/bootstrap.go]: I moved this to the Shell, as it's shell related and a pretty common thing to want to do.

File: bootstrap/bootstrap.go [hunk 2]
```diff
@@ -379,6 +363,11 @@ func (b *Bootstrap) tearDown() error {
 		return err
 	}
 
+	// Support deprecated BUILDKITE_DOCKER* env vars
+	if hasDockerEnvironment(b.shell) {
```
Comment (keithpitt) [inline bootstrap/bootstrap.go]: Can we call it `hasDeprecatedDockerEnvironment` just to make it clear that it shouldn't be considered the future?

Comment (lox) [reply bootstrap/bootstrap.go]: Hah yup, I called it that initially but was worried it was too verbose.

Comment (keithpitt) [reply bootstrap/bootstrap.go]: Verbosity is my fav.

Comment (lox) [reply bootstrap/bootstrap.go]: Ok, done.

**Pull request merged**